src/main/java/picard/analysis/AlignmentSummaryMetricsCollector.java

```diff
@@ -314,7 +313,8 @@ else if (!record.getReadFailsVendorQualityCheckFlag()) {
                                 if (mismatch) hqMismatchCount++;
                             }
 
-                            if (mismatch || SequenceUtil.isNoCall(readBases[readBaseIndex])) {
+                            if (!record.getSupplementaryAlignmentFlag()
```

Comment:
Ditto.

Reply:
If I understand correctly, we can leave this "if" condition for supplementary reads (according to 
yfarjoun comment)?

Reply:
I think that this is right.. @nh13 what do you think?

Reply:
ah no. sorry. this is more complex. Given that we are only looping over the alignment blocks, we need to look at the supplemental reads in order to see the bases....I guess, it's as @nh13 said in the code comment: for metrics that have to do with bases (so the badCycleHistogram is one), we want to see the supplemental reads, but for metrics that have to do with reads, we do not. I think that this means that supplemental reads should be filtered from collectReadData (if supplemental, return) but not from collectQualityData.

Reply:
Thank you for clarifying this! Okay, so I want to make sure that I understand right.. :) I can remove all filtering for supplementary from collectQualityData() , and leave check for secondary in acceptRecord() - this will be necessary logic?

Reply:
I think that's right. @nh13 ?